server/src/modules/slide-objects/slide-objects.service.ts

```diff
@@ -143,6 +147,65 @@ export class SlideObjectsService {
     return deletedSlideObject;
   }
 
+  // WS methods
+
+  async createObject(
+    userId: string,
+    slideId: string,
+    objectProps: ObjectDataProps,
+  ): Promise<SlideObjectResponseObject> {
+    const slide = await this.slidesService.findSlideById(slideId);
+    const user = await this.usersService.findUserById(userId);
+    const createdSlideObject = new this.slideObjectModel({
+      ...objectProps,
+      createdBy: user,
+      slide,
+    });
+    await this.usersService.addSlideObjectToUser(userId, createdSlideObject);
+    await this.slidesService.addSlideObjectToSlide(slideId, createdSlideObject);
+    return createdSlideObject
+      .save()
+      .then(slideObject => this.toResponseSlideObject(slideObject));
+  }
+
+  async updateObject(
+    object: ObjectDataPropsWithId,
+  ): Promise<SlideObjectResponseObject> {
+    const updatedSlideObject = await this.slideObjectModel
+      .findByIdAndUpdate(object._id, object, { new: true })
+      .exec();
+    if (!updatedSlideObject)
+      throw new HttpException('Slide Object not found', HttpStatus.NOT_FOUND);
+    return this.toResponseSlideObject(updatedSlideObject);
+  }
+
+  async deleteObject(
+    userId: string,
+    slideId: string,
+    slideObjectId: string,
+  ): Promise<SlideObjectResponseObject> {
+    await this.slidesService.findSlideById(slideId);
+    await this.usersService.findUserById(userId);
+
+    const deletedSlideObject = await this.slideObjectModel
+      .findByIdAndDelete(slideObjectId)
+      .exec();
+    if (!deletedSlideObject)
+      throw new HttpException('Slide Object not found', HttpStatus.NOT_FOUND);
+
+    await this.usersService.deleteSlideObjectFromUser(
+      userId,
+      deletedSlideObject._id.toString(),
+    );
+    await this.slidesService.deleteObjectFromSlide(
+      slideId,
+      deletedSlideObject._id.toString(),
+    );
+    return this.toResponseSlideObject(deletedSlideObject);
+  }
+
+  // END WS methods
```

Comment:
Maybe it would be better to make separate service for ws?

Reply:
For me it's very confusing that for example we have `deleteSlideObject`, `deleteSlideObjectById` and `deleteObject` methods in the same service.

Reply:
it might turn out half of the methods from [bos].service might be useless (but we can keep them until we can realise what we actually need, they might be good for testing)